src/applications/vaos/routes.jsx

```diff
@@ -1,48 +1,120 @@
+  import(/* webpackChunkName: "vaos-form" */ './containers/ReasonForAppointmentPage').then(
+    m => m.default,
+  ),
+);
+const TypeOfCarePage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/TypeOfCarePage').then(
+    m => m.default,
+  ),
+);
+const CommunityCarePreferencesPage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/CommunityCarePreferencesPage').then(
+    m => m.default,
+  ),
+);
+const TypeOfAudiologyCarePage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/TypeOfAudiologyCarePage').then(
+    m => m.default,
+  ),
+);
+const TypeOfFacilityPage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/TypeOfFacilityPage').then(
+    m => m.default,
+  ),
+);
+const ContactInfoPage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/ContactInfoPage').then(
+    m => m.default,
+  ),
+);
+const VAFacilityPage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/VAFacilityPage').then(
+    m => m.default,
+  ),
+);
+const TypeOfVisitPage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/TypeOfVisitPage').then(
+    m => m.default,
+  ),
+);
+const ReviewPage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/ReviewPage').then(
+    m => m.default,
+  ),
+);
+const ClinicChoicePage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/ClinicChoicePage').then(
+    m => m.default,
+  ),
+);
+const TypeOfSleepCarePage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/TypeOfSleepCarePage').then(
+    m => m.default,
+  ),
+);
+const PreferredDatePage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/PreferredDatePage').then(
+    m => m.default,
+  ),
+);
+const DateTimeRequestPage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/DateTimeRequestPage').then(
+    m => m.default,
+  ),
+);
+const DateTimeSelectPage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/DateTimeSelectPage').then(
+    m => m.default,
+  ),
+);
+const ConfirmationPage = asyncLoader(() =>
+  import(/* webpackChunkName: "vaos-form" */ './containers/ConfirmationPage').then(
+    m => m.default,
+  ),
 );
 
-export default routes;
+export default function createRoutesWithStore(store) {
+  return (
+    <Route path="/" component={VAOSApp}>
+      <IndexRoute component={AppointmentsPage} />
+      <Route
+        path="new-appointment"
+        component={asyncLoader(() =>
+          Promise.all([
+            import(/* webpackChunkName: "vaos-form" */ './components/NewAppointmentLayout'),
+            import(/* webpackChunkName: "vaos-form" */ './reducers/newAppointment'),
+          ]).then(([component, reducer]) => {
+            store.injectReducer('newAppointment', reducer.default);
```

Comment:
Also not super happy injecting a reducer as a side effect of component loading. The other option is to try to do this in the NewAppointmentLayout component itself, but I think I'd have to pull the store from context.

Reply:
Not the biggest fan of injected reducers. What are the down sides of it existing when the store is created?

Reply:
It's a significant chunk of our form code, particularly because the form system does a lot in Redux. So without doing this, a good portion of the form system code, and our form-related app logic, ends up in the main bundle.

What do you dislike about injecting reducers? I don't have much experience lazy loading them, but code splitting seems somewhat ineffective if we're not splitting out the part that has most of the application logic.

Reply:
Got it :D

For dynamic reducers, just adds complexity, which has its trade offs.